src/main/java/com/sap/oss/phosphor/fosstars/model/score/oss/ProjectPopularityScore.java

```diff
@@ -27,114 +24,67 @@
 
   /**
    * Initializes a new score.
    */
   ProjectPopularityScore() {
     super("Open-source project popularity score", DESCRIPTION,
-        NUMBER_OF_GITHUB_STARS, NUMBER_OF_WATCHERS_ON_GITHUB);
+        NUMBER_OF_GITHUB_STARS,
+        NUMBER_OF_WATCHERS_ON_GITHUB,
+        NUMBER_OF_DEPENDENT_PROJECTS_ON_GITHUB);
   }
 
   @Override
   public ScoreValue calculate(Value<?>... values) {
-    Value<Integer> n = findValue(values, NUMBER_OF_GITHUB_STARS,
+    Value<Integer> stars = findValue(values, NUMBER_OF_GITHUB_STARS,
         "Hey! You have to give me a number of stars!");
-    Value<Integer> m = findValue(values, NUMBER_OF_WATCHERS_ON_GITHUB,
+    Value<Integer> watchers = findValue(values, NUMBER_OF_WATCHERS_ON_GITHUB,
         "Hey! You have to give me a number of watchers!");
+    Value<Integer> dependents = findValue(values, NUMBER_OF_DEPENDENT_PROJECTS_ON_GITHUB,
+        "Hey! You have to give me a number of dependents!");
 
-    return scoreValue(starsSubScore(n) + watchersSubScore(m), n, m);
+    return scoreValue(MIN, stars, watchers, dependents)
+        .increase(subScoreFor(stars, BEST_STARS_AMOUNT))
+        .increase(subScoreFor(watchers, BEST_WATCHERS_AMOUNT))
+        .increase(subScoreFor(dependents, BEST_DEPENDENTS_AMOUNT));
   }
 
   /**
-   * Calculates a stars sub-score.
+   * Calculates a sub-score for a value.
    *
-   * @param stars A number of stars.
-   * @throws IllegalArgumentException if a number of stars is negative.
+   * @param value A number of stars.
+   * @param threshold If the value is greater than the threshold,
+   *                  then the max score value is returned.
+   * @throws IllegalArgumentException If the value is negative.
    */
-  private static double starsSubScore(Value<Integer> stars) {
-    if (stars.isUnknown()) {
-      return 0.0;
-    }
-    int n = stars.get();
-    if (n < 0) {
-      throw new IllegalArgumentException("Number of stars is negative! How can it be possible?");
+  private static double subScoreFor(Value<Integer> value, int threshold) {
+    if (value.isUnknown()) {
+      return MIN;
     }
 
-    if (n < BEST_STARS_AMOUNT) {
-      return STARS_SCORE_FACTOR * n;
-    }
-
-    return MAX;
-  }
-
-  /**
-   * Calculates a watchers sub-score.
-   *
-   * @param watchers A number of watchers.
-   * @throws IllegalArgumentException if a number of watchers is negative.
-   */
-  private static double watchersSubScore(Value<Integer> watchers) {
-    if (watchers.isUnknown()) {
-      return 0.0;
-    }
-    int m = watchers.get();
-    if (m < 0) {
-      throw new IllegalArgumentException("Number of watchers is negative! How can it be possible?");
+    int n = value.get();
+    if (n < 0) {
+      throw new IllegalArgumentException("Oops! Expected a non-negative value!");
     }
 
-    if (m < BEST_WATCHERS_AMOUNT) {
-      return WATCHERS_SCORE_FACTOR * m;
+    if (n < threshold) {
+      return MAX / threshold * n;
```

Comment:
If n = 0? Could this lead to some `ArithmeticException`?

Reply:
It should not ideally as you are multiplying n here instead of `n/threshold`. I think it should be fine.

Reply:
> If n = 0? Could this lead to some ArithmeticException?

If n = 0, then `MAX / threshold * n` is 0. I can update it to `MAX * (n / threshold)` if it makes it clearer. The result will be the same.

Reply:
Apologies, my mistake. I believe what you did early was correct. `MAX/threshold * n`. This way it will avoid this `ArithmaticException`
https://www.baeldung.com/java-division-by-zero#integers

Because before this `if` check, you check if `n < 0`

Reply:
@artem-smotrakov do you agree?

Reply:
The previous expression `MAX / threshold * n` and the current one `MAX * n / threshold` do the same and don't throw exceptions because `threshold` can't be currently 0. Neither of these two expressions divide by `n`.